apps/system/utils/utils_free.c

```diff
@@ -56,22 +56,51 @@
 #include <tinyara/config.h>
 #include <stdio.h>
 #include <stdlib.h>
+#if defined(CONFIG_BUILD_PROTECTED) && defined(CONFIG_MM_KERNEL_HEAP)
+#include <fcntl.h>
+#include <errno.h>
+
+#include <tinyara/fs/ioctl.h>
+#include <tinyara/meminfo_drv.h>
+#endif
 
 /****************************************************************************
  * Public Functions
  ****************************************************************************/
 int utils_free(int argc, char **args)
 {
+#if defined(CONFIG_BUILD_PROTECTED) && defined(CONFIG_MM_KERNEL_HEAP)
+	int fd;
+	int ret;
+#endif
 	struct mallinfo data;
 
 #ifdef CONFIG_CAN_PASS_STRUCTS
 	data = mallinfo();
 #else
 	(void)mallinfo(&data);
 #endif
+	printf("              total       used       free    largest\n");
+	printf(" Mem:   %11d%11d%11d%11d\n", data.arena, data.uordblks, data.fordblks, data.mxordblk);
 
+#if defined(CONFIG_BUILD_PROTECTED) && defined(CONFIG_MM_KERNEL_HEAP)
+	fd = open(MEMINFO_DRVPATH, O_RDWR);
+	if (fd < 0) {
+		printf("Open %s Fail, errno %d\n", MEMINFO_DRVPATH, get_errno());
+		return ERROR;
+	}
+
+	ret = ioctl(fd, MEMINFOIOC_KHEAP, (int)&data);
+	if (ret == ERROR) {
+		printf("Meminfo ioctl Fail, errno %d\n", get_errno());
+		close(fd);
+		return ERROR;
+	}
 	printf("              total       used       free    largest\n");
-	printf("Data:   %11d%11d%11d%11d\n", data.arena, data.uordblks, data.fordblks, data.mxordblk);
+	printf("Kmem:	%11d%11d%11d%11d\n", data.arena, data.uordblks, data.fordblks, data.mxordblk);
```

Comment:
For linux compatibility, only with `free` is good for showing the total.
If we want to expand, we'd better use options.